### PR TITLE
Add `cloud-gateway-ips` setting

### PR DESCRIPTION
**Pull request by noahmoss**
Adds a new `cloud-gateway-ips` setting for cloud instances which fetches data from the following store endpoint which redirects to an s3 instance: https://store.metabase.com/static/cloud_gateways.json This will be displayed on the frontend during DB connection. HTTP call results are cached for 5 minutes to avoid frequent requests.

Product doc: https://www.notion.so/metabase/Make-IP-addresses-visible-for-firewall-Cloud-users-4153069c110140e9bd91a164da103584#602785381f30495a81ebb353e0d746ff

### CODE REVIEW

File: src/metabase/public_settings.clj [hunk 1]
```diff
@@ -439,3 +442,29 @@
...
+(def ^:private fetch-cloud-gateway-ips-fn
+  (memoize/ttl
+   (fn []
+       (when (premium-features/is-hosted?)
+         (try
+           (-> (http/get (cloud-gateway-ips-url))
+               :body
+               (json/parse-string keyword)
+               :ip_addresses)
+           (catch clojure.lang.ExceptionInfo e
```
Comment (dpsutton) [inline src/metabase/public_settings.clj:461]: I think this exception class is not a good choice:
```clojure
  (try
    (-> (http/get #_(cloud-gateway-ips-url)
                  "https://storeee.metabase.com/static/cloud_gateways.json")
        :body
        (json/parse-string keyword)
        :ip_addresses)
    (catch clojure.lang.ExceptionInfo e
      :bummer))
Execution error (UnknownHostException) at java.net.Inet6AddressImpl/lookupAllHostAddr (Inet6AddressImpl.java:-2).
storeee.metabase.com: nodename nor servname provided, or not known
```

It doesn't catch all errors. 

```clojure
public-settings=> (type *e)
java.net.UnknownHostException
public-settings=> (ancestors (type *e))
#{java.lang.Exception java.io.Serializable java.lang.Throwable java.lang.Object java.io.IOException}
```
I think we want to catch `Exception` or perhaps `Throwable`. But a clojure specific error class isn't a great choice for interop with so much java like http requests.

Comment (noahmoss) [reply src/metabase/public_settings.clj:461]: I based this off of similar code in `metabase.public-settings.premium-features` that fetches token features from a store endpoint: https://github.com/metabase/metabase/blob/master/src/metabase/public_settings/premium_features.clj#L74

I can definitely change to `Exception` but I'm curious why `clojure.lang.ExceptionInfo` was used there. Maybe that needs to be changed as well.

**Review by dpsutton — APPROVED** [review]
Approving but need to change the exception class that the function catches

File: src/metabase/public_settings.clj (hunk 1 above)
Comment (noahmoss) [reply src/metabase/public_settings.clj:461]: cc @camsaul, seems like the exception caught in `fetch-token-status*` was changed from `Throwable` to `clojure.lang.ExceptionInfo` this PR: https://github.com/metabase/metabase/pull/13590/files#diff-1b2e7bb52ebe54644c80a90c2ba72ffb66d43ef4a1d042c7f587d6b7b49557dcR76

Was there a reason for this?

**Comment by ranquild** [comment]
Part of epic https://github.com/metabase/metabase/pull/18931

**Pull request merged**